src/services/messagesService.ts

```diff
@@ -208,39 +218,56 @@ export default class MessagesService {
         )
         // Enrich the message with legal_message retrieved from pec-server
         .chain(message =>
-          new TE.TaskEither(
-            new T.Task(() =>
-              // FIXME: add JWT bearer implmenetation
-              this.pecClient.getClient("dummy-bearer-token").getMessage({
-                id: message.content.legal_data?.message_unique_id || "ERROR" // Workaround: legal_data esistence has already be checked
-              })
+          new TokenService()
+            .getPecServerToken(
+              message.fiscal_code,
+              PEC_SERVER_TOKEN_SECRET,
+              PEC_SERVER_TOKEN_EXPIRATION,
+              PEC_SERVER_TOKEN_ISSUER
             )
-              .map(response =>
-                response.mapLeft(_ =>
-                  ResponseErrorInternal(
-                    "Error decoding pecServer getMessage response"
-                  )
-                )
+            .mapLeft(e =>
+              ResponseErrorInternal(
+                `Error computing the PEC Server JWT: ${e.message}`
               )
-              .map(response =>
-                response.map(
-                  E.fromPredicate(isPecServerGetMessageSuccess, e =>
-                    ResponseErrorInternal(
-                      `Error getting the message from pecServer getMessage endpoint (received a ${e.status})` // TODO: disjoint the errors
+            )
+            .chain(pecServerJwt =>
+              new TE.TaskEither(
+                new T.Task(() =>
+                  this.pecClient.getClient(pecServerJwt).getMessage({
+                    id:
+                      message.content.legal_data?.message_unique_id ||
+                      youShouldNotBeHere(
+                        "legal_data esistence has already been checked"
+                      )
+                  })
+                )
+                  .map(response =>
+                    response.mapLeft(_ =>
```

Comment:
you can use `bimap()`

Reply:
refactored to avoid the nested map. thank you!